frontend-react/src/components/Admin/EditSenderSettings.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { Suspense } from "react";
+import { Button, GridContainer, Grid } from "@trussworks/react-uswds";
+import { useResource, NetworkErrorBoundary, useController } from "rest-hooks";
+import { RouteComponentProps, useHistory } from "react-router-dom";
+
+import { ErrorPage } from "../../pages/error/ErrorPage";
+import OrgSenderSettingsResource from "../../resources/OrgSenderSettingsResource";
+import { showAlertNotification, showError } from "../AlertNotifications";
+
+import { TextInputComponent } from "./AdminFormEdit";
+
+type Props = { orgname: string; sendername: string; action: string };
+
+export function EditSenderSettings({ match }: RouteComponentProps<Props>) {
+    const orgname = match?.params?.orgname || "";
+    const sendername = match?.params?.sendername || "";
+    const action = match?.params?.action || "";
+    const history = useHistory();
+
+    const FormComponent = () => {
+        const orgSenderSettings: OrgSenderSettingsResource = useResource(
+            OrgSenderSettingsResource.detail(),
+            { orgname, sendername: sendername, action: action }
+        );
+
+        const { fetch: fetchController } = useController();
+        const saveSenderData = async () => {
+            switch (action) {
+                case "edit":
+                    try {
+                        console.log("EDIT SETTING");
+                        const data = JSON.stringify(orgSenderSettings);
+                        await fetchController(
+                            OrgSenderSettingsResource.update(),
+                            { orgname, sendername: sendername },
+                            data
+                        );
+                        showAlertNotification(
+                            "success",
+                            `Item '${sendername}' has been updated`
+                        );
+                        history.goBack();
+                    } catch (e: any) {
+                        console.trace(e);
+
+                        showError(
+                            `Updating item '${sendername}' failed. ${e.toString()}`
+                        );
+                        return false;
+                    }
+                    break;
+                case "clone":
+                    try {
+                        const data = JSON.stringify(orgSenderSettings);
+                        await fetchController(
+                            OrgSenderSettingsResource.update(),
+                            { orgname, sendername: orgSenderSettings.name },
+                            data
+                        );
```

Comment:
Couple things (applies to both sender and receiver cloning):
1. I think we should always set the `createdBy` and `createdAt` to the current user/time when cloning. otherwise these will be set to the values of the cloned sender/receiver
2. Calling `OrgSenderSettingsResource.update()` feels a little unintuitive to me. The way the code reads makes me think that we are updating the resource we are trying to clone. It seems to work fine in testing so maybe not a big deal but I'm wondering if there is a better convention?

Reply:
1. Agreed. I _think_ the server side of the API should enforce timestamps (if not, likely a bug), but if not we should definitely follow up.
2. This is how the Prime REST API is done. It's an Upsert combining new and update. But I agree it's confusing and if nothing else, we should document it in the comments.

Reply:
@TomNUSDS added that comment. @sean-usds the API _is_ actually creating a new timestamp on cloning. I think it might've appeared that way due to the [cloning bug](https://app.zenhub.com/workspaces/experience-607d9d5e68b95200150fec37/issues/cdcgov/prime-reportstream/3934). 

before:
![createdAt update CSV](https://user-images.githubusercontent.com/92405130/150438702-a5587ff6-44de-4628-b6e5-1b404a99391c.PNG)

after:
![createdAt update CSV_CLONE](https://user-images.githubusercontent.com/92405130/150438722-1bbf357f-8a1a-4efc-8547-caba9e74a097.PNG)